modules/mono/editor/GodotTools/GodotTools.Core/FileUtils.cs

```diff
@@ -0,0 +1,27 @@
+using System.IO;
+
+namespace GodotTools.Core
+{
+    public static class FileUtils
+    {
+        public static void SaveBackupCopy(string filePath)
+        {
+            string backupPathBase = filePath + ".old";
+            string backupPath = backupPathBase;
+
+            const int maxAttempts = 5;
+            int attempt = 0;
+
+            while (File.Exists(backupPath) && attempt < maxAttempts)
+            {
+                backupPath = backupPathBase + "." + (attempt + 1);
```

Comment:
Could be simplified to not have the `+ 1` either by changing the initial value to 1 or by moving the `++` line above this one.